.github/workflows/auto-draft-release.yml

```
@@ -0,0 +1,57 @@
+name: Automatically Draft a Release on Version Bump
+
+on:
+  push:
+    branches:
+      - main
+
+jobs:
+  create-tag:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write  # Needed to create tags
+
+    steps:
+      # Check out the repository
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0  # Fetch all history to ensure we can check if tag exists
+
+      - name: Read VERSION file
+        id: version
+        run: |
+          # Use grep to filter out comment lines, then take the first non-comment line
+          VERSION_VALUE=$(grep -v '^#' VERSION | head -n 1 | tr -d '[:space:]')
+          echo "Detected version: $VERSION_VALUE"
+          echo "version=$VERSION_VALUE" >> $GITHUB_OUTPUT
+
+      - name: Check if tag exists
+        id: check_tag
+        run: |
+          if git tag -l "v${{ steps.version.outputs.version }}" | grep -q .; then
+            echo "tag_exists=true" >> $GITHUB_OUTPUT
+          else
+            echo "tag_exists=false" >> $GITHUB_OUTPUT
+          fi
+
+      # Create and push the tag if it doesn't already exist
+      - name: Create and push tag
+        if: steps.check_tag.outputs.tag_exists == 'false'
+        run: |
+          git config user.name "GitHub Actions Bot"
+          git config user.email "actions@github.com"
+          git tag -a v${{ steps.version.outputs.version }} -m "Version ${{ steps.version.outputs.version }}"
+          git push origin "v${{ steps.version.outputs.version }}"
+
+      - name: Create draft GitHub Release
+        if: steps.check_tag.outputs.tag_exists == 'false'
+        uses: softprops/action-gh-release@v1
```

```suggestion
        uses: softprops/action-gh-release@v2
```
Maybe we should use v2; that seems to be the latest version: https://github.com/softprops/action-gh-release/releases.